pip/req/req_set.py

```diff
@@ -447,6 +474,12 @@ def _prepare_file(self, finder, req_to_install):
                 if self.is_download:
                     req_to_install.archive(self.download_dir)
             elif req_to_install.satisfied_by:
+                if require_hashes:
+                    logger.info(
+                        'Since it is already installed, we are trusting this '
+                        'package without checking its hash. To ensure a '
+                        'completely repeatable environment, install into an '
+                        'empty virtualenv.')
```

Comment:
I consider installing into a non-empty venv inadvisable for production deployments—you'll never get rid of packages you've removed from reqs.txt, and they might cause weird behavior through monkeypatching or entrypoints or what-have-you. (Take, for example, my evil https://pypi.python.org/pypi/tracefront/, which masquerades as a stdlib package to get itself loaded.) This message is meant to scare people away from that practice. But I could understand updating a venv on one's dev machine. With my motivation better spelled out, what do you think? INFO or DEBUG? (Ideally, I think we'd roll up all such reports into a single INFO message and report it all at once, but I don't relish doing that refactoring atm.)